hr_attendance_missing_days/models/hr_employee.py

```diff
@@ -0,0 +1,72 @@
+# © 2023 initOS GmbH
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from datetime import datetime, time
+
+import pytz
+
+from odoo import models
+
+_logger = logging.getLogger(__name__)
+
+
+def ensure_tz(dt, tz=None):
+    if not dt.tzinfo:
+        dt = pytz.utc.localize(dt)
+    return dt.astimezone(tz) if tz else dt
+
+
+class Employee(models.Model):
+    _inherit = "hr.employee"
+
+    def create_missing_attendances(self, date_from=None, date_to=None):
+        for emp in self.search([]):
+            emp._create_missing_attendances(date_from, date_to)
+
+    def _create_missing_attendances(self, date_from=None, date_to=None):
+        self.ensure_one()
+
+        reason = self.env.company.sudo().attendance_missing_days_reason
+        if not reason:
+            return
+
+        if not date_from:
+            date_from = datetime.combine(
+                self.env.company.sudo().overtime_start_date, time.min
+            )
+
+        if not date_to:
+            date_to = datetime.now()
+
+        date_from, date_to = map(ensure_tz, (date_from, date_to))
```

Comment:
I expect the way you localize date_from to cause trouble for people working at fringe times: If overtime start date is 2023-08-14 and the employee's timezone Europe/Amsterdam, we want date_from to be 2023-08-13 22:00:00 UTC, but here we get 2023-08-14 00:00:00 UTC.

Currently you don't use the employee's timezone at all here, but when you do, you'll have to localize the other way around:

```python
pytz.timezone(employee.tz).localize(date_from)
```

Reply:
I added a unit tests to cover some edge cases like 23:30 in various tz. It works fine. Be aware that the actual times are coming from the resource.calendar `_work_intervals_batch` and are already localized. `date_from` and `date_to` are just the time frame to calculate (if you only want to partially invalidate those).